DataProcessing/utils.py

```diff
@@ -91,7 +93,54 @@ def trim_video(input_path, output_path, limit):
     temp_dir.cleanup()
 
 
-if __name__ == '__main__':
-    trim_video('/home/bar_cohen/Data-Shoham/1.8.21_cam1/videos/IPCamera_20210801095724.avi',
-               '/home/bar_cohen/KinderGuardian/Videos/trimmed_1.8.21-095724.mp4', limit=500)
+def viz_data_on_video(input_vid, output_path, pre_labeled_pkl_path=None,path_to_crops=None):
+    """
+    This func assumes that the input video has been run by the track and reid model data creator to
+    create a pre-annoted set.
+    Args:
+        input_vid:
+        pre_labeled_pkl_path:
+
+    Returns:
 
+    """
+    assert pre_labeled_pkl_path or path_to_crops , "You must enter either a pkl to cropsDB or the crops folder"
+    crops = None
+    if path_to_crops:
+        assert os.path.isdir(path_to_crops) , "Path must be a CropDB folder"
+        crops = []
+        for file in os.listdir(path_to_crops):
+            crop_path = os.path.join(path_to_crops, file)
+            _ , extention = os.path.splitext(crop_path)
+            if os.path.split(crop_path)[-1][0] != 'Face' and extention in ['.jpg', '.png']: # skip face crops for viz
+                crops.append(create_Crop_from_str(crop_path))
+    elif pre_labeled_pkl_path:
+        assert os.path.isfile(pre_labeled_pkl_path) , "Path must be a CropDB file"
+        crops = pickle.load(open(pre_labeled_pkl_path, 'rb'))
+
+    # create frame_to_crops dict
+    crop_dict_by_frame = defaultdict(list)
+    for crop in crops:
+        crop_dict_by_frame[crop.frame_id].append(crop)
+
+    imgs = mmcv.VideoReader(input_vid)
+    assert len(imgs) == len(crop_dict_by_frame.keys()) , "Must have as many cropped_frames as crops"
+    temp_dir = tempfile.TemporaryDirectory()
+    temp_path = temp_dir.name
+    fps = int(imgs.fps)
+
+    for i,frame in enumerate(imgs):
+        cur_crops = crop_dict_by_frame[i]
```

Comment:
In case there are no crops for a given frame (like in the case described above) this will raise a `KeyError` exception.
To avoid this, change to `crop_dict_by_frame.get(i)` which will return None if there is no such key instead of an exception. Generally, it is always better to access dict keys with `dict.get(key)` to avoid this error. See [why-dict-getkey-instead-of-dictke](https://stackoverflow.com/questions/11041405/why-dict-getkey-instead-of-dictkey)

Reply:
Agree, and fit for the solution to the prev comment, thanks!